state/upgrades_test.go

```diff
@@ -2760,6 +2763,55 @@ func (s *upgradesSuite) TestRemoveUseFloatingIPConfigFalse(c *gc.C) {
 	c.Assert(ok, jc.IsFalse)
 }
 
+func (s *upgradesSuite) TestMigrateCappedTxnsLogCollection(c *gc.C) {
+	coll, resCloser := s.state.db().GetRawCollection(txnLogC)
+	defer resCloser()
+
+	err := coll.DropCollection()
+	c.Assert(err, jc.ErrorIsNil)
+
+	err = createCollection(coll, &mgo.CollectionInfo{
+		Capped:   true,
+		MaxBytes: 1024 * 1024,
+		MaxDocs:  1,
+	})
+	c.Assert(err, jc.ErrorIsNil)
+
+	err = coll.Insert(bson.D{{Name: "_id", Value: "0"}})
+	c.Assert(err, jc.ErrorIsNil)
+	err = coll.Insert(bson.D{{Name: "_id", Value: "1"}})
+	c.Assert(err, jc.ErrorIsNil)
+	err = coll.Insert(bson.D{{Name: "_id", Value: "2"}})
+	c.Assert(err, jc.ErrorIsNil)
+
+	n, err := coll.Count()
+	c.Assert(err, jc.ErrorIsNil)
+	c.Assert(n, gc.Equals, 1)
+
+	s.assertData(c, upgradedData(coll, bsonMById{{
```

Comment:
Do we need to assert the collection is now uncapped

Reply:
a lot more code, it effectively does that by testing capped count, then testing it doesn't have the capped count.

I'll try just a transaction before and after.